lib/Yield/Support/GraphLayout/SugiyamaStyle/GraphPreparation.cpp

```diff
@@ -213,15 +220,15 @@ RankContainer partitionLongEdges(InternalGraph &Graph,
         revng_assert(Edge != nullptr);
         SavedLongEdges.emplace_back(From, To, std::move(*Edge));
         Iterator = From->removeSuccessor(Iterator);
+        if (!To->hasPredecessors())
+          GraphEntry->addSuccessor(To, InternalNode::Edge(Graph));
       } else {
         ++Iterator;
       }
     }
   }
 
   // Calculate real ranks for the remainder of the graph.
-  ensureSingleEntry(Graph, &Ranks);
-  revng_assert(HasSingleEntryPoint(Graph));
   Ranks = rankNodes<Strategy>(Graph.getEntryNode());
```

Comment:
Here and later in this function you call `Graph.getEntryNode`. Now that you have the `GraphEntry` guard, wouldn't it make more sense to use that instead of calling `Graph.getEntryNode()`?

Reply:
Whoops, I didn't pay attention to this, I guess. But also consider that we still have to use `setEntryNode` within the `GraphEntry` no matter what. `getEntryPoints` from `GraphAlgorithms.h` uses graph-level iteration - i. e. `GenericGraph`'s `llvm::GraphTraits`, which relies on the `getEntryNode`. As such, there's no avoiding it one way or the other.

Reply:
Note that the guard destructor even does `revng_assert(Graph.getEntryNode() == &VirtualEntry)`.

Reply:
Either way, I replaced them.